src/app/clusters/time-synchronization-server/time-synchronization-server.h

```diff
@@ -96,20 +110,53 @@ class TimeSynchronizationServer : public FabricTable::Delegate
     void ClearEventFlag(TimeSyncEventFlag flag);
 
     // Fabric Table delegate functions
-    void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex);
+    void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex) override;
+
+#if TIME_SYNC_ENABLE_TSC_FEATURE
+    // CASE connection functions
+    void OnDeviceConnectedFn(Messaging::ExchangeManager & exchangeMgr, const SessionHandle & sessionHandle);
+    void OnDeviceConnectionFailureFn();
+
+    // AttributeCache::Callback functions
+    void OnAttributeChanged(ClusterStateCache * cache, const ConcreteAttributePath & path) override {}
+    void OnDone(ReadClient * apReadClient) override;
+#endif
+
+    // Platform event handler functions
+    void OnPlatformEventFn(const DeviceLayer::ChipDeviceEvent & event);
+
+    void OnTimeSyncCompletionFn(TimeSourceEnum timeSource, GranularityEnum granularity);
+    void OnFallbackNTPCompletionFn(bool timeSyncSuccessful);
 
 private:
+    static constexpr size_t kMaxDefaultNTPSize = 128;
     DataModel::Nullable<Structs::TrustedTimeSourceStruct::Type> mTrustedTimeSource;
     TimeSyncDataProvider::TimeZoneObj mTimeZoneObj{ Span<TimeSyncDataProvider::TimeZoneStore>(mTz), 0 };
     TimeSyncDataProvider::DSTOffsetObj mDstOffsetObj{ DataModel::List<Structs::DSTOffsetStruct::Type>(mDst), 0 };
-    GranularityEnum mGranularity;
+    GranularityEnum mGranularity = GranularityEnum::kNoTimeGranularity;
 
     TimeSyncDataProvider::TimeZoneStore mTz[CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE];
     Structs::DSTOffsetStruct::Type mDst[CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE];
 
     TimeSyncDataProvider mTimeSyncDataProvider;
     static TimeSynchronizationServer sTimeSyncInstance;
     TimeSyncEventFlag mEventFlag = TimeSyncEventFlag::kNone;
+#if TIME_SYNC_ENABLE_TSC_FEATURE
+    Platform::UniquePtr<app::ClusterStateCache> mAttributeCache;
```

Comment:
So once we read the values... why do we hold on to them forever?  Again, this is not great for constrained devices...

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/29396